### PR TITLE
be more accurate when checking if cosmos enabled

### PR DESCRIPTION
**Pull request by tamarrow**
#525

### CODE REVIEW

File: dcos/cosmospackage.py [hunk 1]
```diff
@@ -37,10 +37,14 @@ def enabled(self):
         # authorized for the command specified, not this endpoint
         except DCOSAuthorizationException:
             return True
-        # all other errors are treated as endpoint not available
+        # allow exception through so we can show user actual http exception
+        # except 404, because then the url is fine, just not cosmos enabled
         except Exception as e:
             logger.exception(e)
-            return False
+            if "HTTP 404: Not Found" in str(e):
```
Comment (jsancio) [inline dcos/cosmospackage.py]: Okay. For my own edification, I take it that requests doesn't return a typed exception with the status code when it is 404. @tamarrow can we confirm?

Comment (tamarrow) [inline dcos/cosmospackage.py]: nope: https://github.com/mesosphere/dcos-cli/blob/master/dcos/http.py#L222
screen shot 2016-04-05 at 5 58 46 pm

Comment (jsancio) [inline dcos/cosmospackage.py]: We can fix that code, no? We have the response. We can add a field to DCOSHTTPException for the status.

Comment (tamarrow) [inline dcos/cosmospackage.py]: yep! updated

**Comment by jsancio** [comment]
LGTM.

**Pull request merged**